holoviews/core/dimension.py

```diff
@@ -1077,13 +1077,24 @@ def __unicode__(self):
         return unicode(PrettyPrinter.pprint(self))
 
 
-
     def __call__(self, options=None, **kwargs):
+        return self.opts(options, **kwargs)
+
+    def opts(self, options=None, **kwargs):
         """
         Apply the supplied options to a clone of the object which is
         then returned. Note that if no options are supplied at all,
         all ids are reset.
         """
+        from ..util.parser import OptsSpec
+        if isinstance(options, basestring):
+            try:
+                options = OptsSpec.parse(options)
+            except:
+                options = OptsSpec.parse(
+                    '{clsname} {options}'.format(clsname=self.__class__.__name__,
+                                                 options=options))
```

Comment:
Sure. This falls under my bullet points for something to improve. The exception is part of it, but the whole ``try/except`` here isn't a very robust approach. I can specify an exception now but this whole thing needs improving (in another PR imho).